plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -180,6 +215,27 @@ StorageOptions createStorageOptions(final GoogleCloudStorageClientSettings clien
         return storageOptionsBuilder.build();
     }
 
+    @SuppressForbidden(reason = "ok to open connection here")
+    private static String getDefaultProjectId() throws IOException {
+        String metaHost = System.getenv("GCE_METADATA_HOST");
```

Comment:
I was wondering if there is ever a case where the [metadata server](https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying) is not `metadata.google.internal`, but I suppose a) there is no guarantee this will remain for ever the same b) it might be useful esp. under additional layers under k8s etc.

Reply:
This is purely used for testing. While the Google library uses this environment variable in other places (e.g. to load the credentials from the environment) it does not check it when loading the project id :/